src/neo/SmartContract/Native/Tokens/Nep5Token.cs

```diff
@@ -68,7 +68,21 @@ internal protected virtual void Mint(ApplicationEngine engine, UInt160 account,
             state.Balance += amount;
             storage = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_TotalSupply), () => new StorageItem(BigInteger.Zero));
             storage.Add(amount);
-            engine.SendNotification(Hash, "Transfer", new Array { StackItem.Null, account.ToArray(), amount });
+            onPostTransfer(engine, null, account, amount);
+        }
+
+        private void onPostTransfer(ApplicationEngine engine, UInt160 from, UInt160 to, BigInteger amount)
+        {
+            // Send notification
+
+            engine.SendNotification(Hash, "Transfer",
+                new Array { from == null ? StackItem.Null : from.ToArray(), to == null ? StackItem.Null : to.ToArray(), amount });
+
+            if (!engine.ContractExists(to, "onPayment")) return;
```

Comment:
Hi Shargon, I noticed that now neo is doing some method checks using hardcoded strings for method names ('verify', for example). I believe this may cause some issues in the future. First 'verify', now 'onPayment'. Maybe the manifest file should have a section to map these methods into ABI methods. This way, instead of checking only if the method exists, you would get the name of it first, and if it exists,  execute it.

Reply:
> instead of checking only if the method exists, you would get the name of it first, and if it exists, execute it.

Could you give me an example, i don't see the difference right now. 😕

Reply:
@lock9